extensions/artemis-core/runtime/src/main/java/io/quarkus/artemis/core/runtime/ArtemisCoreRecorder.java

```diff
@@ -1,12 +1,22 @@
 package io.quarkus.artemis.core.runtime;
 
-import io.quarkus.arc.runtime.BeanContainer;
+import java.util.function.Supplier;
+
+import org.apache.activemq.artemis.api.core.client.ActiveMQClient;
+import org.apache.activemq.artemis.api.core.client.ServerLocator;
+
 import io.quarkus.runtime.annotations.Recorder;
 
 @Recorder
 public class ArtemisCoreRecorder {
 
-    public void setConfig(ArtemisRuntimeConfig config, BeanContainer container) {
-        container.instance(ArtemisCoreProducer.class).setConfig(config);
+    public Supplier<ServerLocator> getServerLocatorSupplier(ArtemisRuntimeConfig config) {
+        return () -> {
+            try {
+                return ActiveMQClient.createServerLocator(config.url);
+            } catch (Exception e) {
+                throw new RuntimeException("Could not create ServerLocator", e);
+            }
+        };
```

Comment:
Thanks!

We prefer to avoid lambdas in runtime code as they have a memory cost. Could you adjust this one and the one below?